src/core/NotePlayHandle.cpp

```diff
@@ -29,9 +29,11 @@
 #include "AudioEngine.h"
 #include "BasicFilters.h"
 #include "DetuningHelper.h"
+#include "GuiApplication.h"
 #include "InstrumentSoundShaping.h"
 #include "InstrumentTrack.h"
 #include "Instrument.h"
+#include "PianoRoll.h"
```

Comment:
Since this is a core file, we will not want to add GuiApplication.h or PianoRoll.h here.

Reply:
Is it acceptable to add them to InstrumentTrack.cpp?

Reply:
For now it might be, but if we can find a way not doing it we should. I know it'll make it more complicated, but we want to try to keep core and GUI as separate as we can.